pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -242,17 +248,40 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env *model.Environme
 				filter.ConfigType = &listener.Filter_Config{Config: util.MessageToStruct(tcpProxy)}
 			}
 
+			filterChains := []listener.FilterChain{
+				{
+					Filters: []listener.Filter{filter},
+				},
+			}
+
+			// The virtual listener will handle all traffic that does not match any other listeners, and will
+			// blackhole/passthrough depending on the outbound traffic policy. When passthrough is enabled,
+			// this has the risk of triggering infinite loops when requests are sent to the pod's IP, as it will
+			// send requests to itself. To block this we add an additional filter chain before that will always blackhole.
+			if pilot.RestrictPodIPTrafficLoops.Get() {
+				var cidrRanges []*core.CidrRange
+				for _, ip := range node.IPAddresses {
+					cidrRanges = append(cidrRanges, util.ConvertAddressToCidr(ip))
+				}
+				blackhole := blackholeStructMarshalling
+				if util.IsXDSMarshalingToAnyEnabled(node) {
+					blackhole = blackholeAnyMarshalling
+				}
+				filterChains = append([]listener.FilterChain{{
```

Comment:
This is confusing. You are adding single filter chain. 
Why not `filterChains = append(filterChains, listener.FilterChain {... } ])` ?

Reply:
I am prepending this, not appending

Reply:
I didn't realize it.. 
Notes that the filter chains will be applied as unordered at envoy proxy.